File: back/src/main/java/com/triptrace/domain/auth/auth/repository/EmailVerificationRepository.java
package com.triptrace.domain.auth.auth.repository;

import com.triptrace.domain.auth.auth.entity.EmailVerification;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface EmailVerificationRepository extends JpaRepository<EmailVerification, Long> {

    Optional<EmailVerification> findTopByEmailOrderByCreatedAtDesc(String email);
}

File: back/src/main/java/com/triptrace/domain/auth/auth/repository/EmailVerificationRepository.kt
package com.triptrace.domain.auth.auth.repository

import com.triptrace.domain.auth.auth.entity.EmailVerification
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface EmailVerificationRepository : JpaRepository<EmailVerification, Long> {

    fun findTopByEmailOrderByCreatedAtDesc(email: String): Optional<EmailVerification>
}
